### PR TITLE
Remove backup and rollback from Linux and Mac upgrade scripts

### PR DESCRIPTION
**Pull request by jr0me**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/22522|

## Description

In this PR, backup and rollback logic is removed from the agent upgrade scripts.
With this simplification, both Linux and Mac can use the same script.
Using the same script still requires a change on Wazuh-Packages' side: https://github.com/wazuh/wazuh-packages/pull/2885

## Tests

- [ ] Source upgrade
- [ ] Package upgrade

### CODE REVIEW

File: src/init/pkg_installer.sh [hunk 1]
```diff
@@ -1,194 +1,67 @@
...
+        if command -v dpkg >/dev/null 2>&1; then
+            dpkg -i ./var/upgrade/wazuh-agent* >> ./logs/upgrade.log 2>&1
         else
-            echo "$(date +"%Y/%m/%d %H:%M:%S") - Wazuh SELinux module installation skipped (SELinux is disabled)." >> ./logs/upgrade.log
+            echo "$(date +"%Y/%m/%d %H:%M:%S") - Upgrade failed. DEB package found but dpkg command not found." >> ./logs/upgrade.log
+            echo -ne "2" > ./var/upgrade/upgrade_result
+            rm -f $LOCK
+            exit 1
+        fi
+    else
```
Comment (TomasTurina) [inline src/init/pkg_installer.sh:47]: Check https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-linux.html. Be aware we should also support `apk` packages.

File: src/win32/do_upgrade.ps1
Comment (TomasTurina) [inline src/win32/do_upgrade.ps1:1]: I still see lots of functions here, for example stop agent. Shouldn't these be handled by the MSI itself?

**Comment by TomasTurina** [comment]
Rebase your branch to fix the failing IT on windows.

File: src/init/pkg_installer.sh (hunk 1 above)
Comment (jr0me) [reply src/init/pkg_installer.sh:56]: Added in 08f1fbde55339e4fad0565a97f8399e464d40bd9

File: src/win32/do_upgrade.ps1
Comment (jr0me) [reply src/win32/do_upgrade.ps1:1]: I removed the stop function in ec7470b57f5bf4b041ea5f79a0a7ea234fe936ad the others seem to be necessary.

**Comment by jr0me** [comment]
> Rebase your branch to fix the failing IT on windows.

Both branches were already at the latest commit from 4.9.0 ...

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**